apps/user-portal/src/pages/user-profile.tsx

```diff
@@ -17,58 +17,281 @@
  */
 
 import * as React from "react";
-import { Link } from "react-router-dom";
-import { Button, Container, Divider, Form, Grid, Header } from "semantic-ui-react";
-import { getProfileInfo } from "../actions/profile";
+import { Button, Container, Divider, Form, Grid, Header, Icon, Message, Segment } from "semantic-ui-react";
+import { getProfileInfo, updateProfileInfo } from "../actions/profile";
 import { UserImagePlaceHolder } from "../components";
 import { InnerPageLayout } from "../layouts";
 import { createEmptyProfile } from "../models/profile";
 
+/**
+ * User Profile Page of the User Portal
+ */
 export class UserProfilePage extends React.Component<any, any> {
+    /**
+     * constructor
+     * @param props
+     */
     constructor(props) {
         super(props);
         this.state = createEmptyProfile();
+        this.handleSave = this.handleSave.bind(this);
+        this.handleFieldChange = this.handleFieldChange.bind(this);
+        this.handleCancel = this.handleCancel.bind(this);
+        this.handleEdit = this.handleEdit.bind(this);
     }
 
     public componentWillMount() {
         getProfileInfo()
-        .then((response) => {
+            .then((response) => {
                 this.setProfileDetails(response);
             });
     }
 
+    /**
+     * The following method handles the change of state of the input fields.
+     * The id of the event target will be used to set the state.
+     * @param event
+     */
+    public handleFieldChange = (event) => {
+        this.setState({
+            [event.target.id]: event.target.value
+        });
+        event.preventDefault();
+    }
+
+    /**
+     * The following method handles the onClick event of the edit button
+     * @param event
+     */
+    public handleEdit = (event) => {
+        this.setState({
+            [event.target.id]: true
+        });
+    }
+
+    /**
+     * The following method handles the onClick event of the cancel button
+     * @param event
+     */
+    public handleCancel = (event) => {
+        this.setState({
+            [event.target.id]: false
+        });
+    }
+
+    /**
+     * The following method handles the onClick event of the save button
+     * The update request will be sent depending on the id of the event target
+     * @param event
+     */
+    public handleSave = (event) => {
+        const data = {
+            schemas: [
+                "urn:ietf:params:scim:api:messages:2.0:PatchOp"
+            ],
+            Operations: [
```

Comment:
make this all simple

Reply:
The value of the key seems to be case sensitive. The request is not sent successfully when we change the key to lower case. So is it okay if we keep it this way?